ruby.js

```diff
@@ -104,6 +106,51 @@ function balanceEvent(event) {
 	if (event && event.document) balancePairs(event.document);
 }
 
+function completeCommand(args) {
+	if (process.platform == 'win32')
+		return cp.spawn('cmd', ['/c', 'rct-complete'].concat(args));
+	return cp.spawn('rct-complete', args);
+}
+
+function completionProvider(document, position, token) {
+	return new Promise((resolve, reject) => {
+		const line = position.line + 1;
+		const column = position.character;
+
+		var child = completeCommand([
+			'--completion-class-info',
+			'--dev',
+			'--fork',
+			'--line='+line,
+			'--column='+column]);
+
+		var outbuf = [], errbuf = [];
+		child.stderr.on('data', (data) => errbuf.push(data.toString()));
+		child.stdout.on('data', (data) => outbuf.push(data.toString()));
+		child.stdout.on('end', () => {
+			if (errbuf.length > 0) reject(errbuf.join(''));
```

Comment:
This will still cause a double call of one of the promise returns. If you really want to handle the `stderr` output this way, you'll have to `return` from here.